libr/arch/p/gb/plugin.c

```diff
@@ -1572,25 +1582,45 @@ static int archinfo(RAnal *anal, int q) {
 	return -1;
 }
 
-RAnalPlugin r_anal_plugin_gb = {
-	.name = "gb",
-	.desc = "Gameboy CPU code analysis plugin",
-	.license = "LGPL3",
+static bool esil_cb(RArchSession *as, RArchEsilAction action) {
+	REsil *esil = as->arch->esil;
+	if (!esil) {
+		return false;
+	}
+
+	switch (action) {
+	case R_ARCH_ESIL_INIT:
+		esil_gb_init (esil);
+		break;
+	case R_ARCH_ESIL_FINI:
+		esil_gb_fini (esil);
+		break;
+	default:
+		return false;
+	}
+
+	return true;
+}
+
+RArchPlugin r_arch_plugin_gb = {
+	.meta = {
+		.name = "gb",
+		.desc = "Gameboy CPU code analysis plugin",
+		.license = "LGPL3",
+	},
 	.arch = "z80",
-	.esil = true,
-	.bits = 16,
-	.op = &gb_anop,
-	.opasm = &gb_opasm,
-	.set_reg_profile = &set_reg_profile,
-	.archinfo = archinfo,
-	.esil_init = esil_gb_init,
-	.esil_fini = esil_gb_fini,
+	.info = info,
+	.regs = regs,
+	.bits = R_SYS_BITS_PACK1 (16),
+	.decode = decode,
+	.encode = encode,
+	.esilcb = esil_cb,
 };
 
 #ifndef R2_PLUGIN_INCORE
 R_API RLibStruct radare_plugin = {
 	.type = R_LIB_TYPE_ANAL,
```

Comment:
```suggestion
	.type = R_LIB_TYPE_ARCH,
```

Reply:
Oops. Went too fast and didn't check with lint.sh